http/file_server.ts

```diff
@@ -118,28 +124,34 @@ export async function serveFile(
     headers.set("date", date.toUTCString());
   }
 
-  // Set last modified header if last modification timestamp is available
-  if (fileInfo.mtime instanceof Date) {
-    const lastModified = new Date(fileInfo.mtime);
-    headers.set("last-modified", lastModified.toUTCString());
-
-    // Create a simple etag that is an md5 of the last modified date and filesize concatenated
-    const etag = toHashString(
+  // Create a simple etag that is an md5 of the last modified date and filesize concatenated
+  const etag = fileInfo.mtime
+    ? toHashString(
       await createHash(
         etagAlgorithm ?? "FNV32A",
-        `${lastModified.toJSON()}${fileInfo.size}`,
+        `${fileInfo.mtime.toJSON()}${fileInfo.size}`,
       ),
-    );
+    )
+    : DENO_DEPLOYMENT_ID;
```

Comment:
Please also hash the deployment id.

Reply:
I changed it like that. For reference, can you tell me why I should do that?

Reply:
I guess the purpose is not to reveal `DENO_DEPLOYMENT_ID` to the visitors. Exposing `DENO_DEPLOYMENT_ID` is not directly exploitable/abusable, but if the attacker keeps memorizing them and find vulnerability in some old version of the deployment, then they can use old `DENO_DEPLOYMENT_ID` to invoke 'old' version of deployment by accessing `https://project-name--<that DENO_DEPLOYMENT_ID>.deno.dev/`

The current way of transforming `DENO_DEPLOYMENT_ID` is reversible. I think we should apply `createHash` before pass it to `toHashString`

Reply:
It makes a lot of sense. thank you!

Another question. The hashing algorithm for the deployment id is currently fixed to FNV32A. Should the `etagAlgorithm` optionally passed by the user be honored here?

Reply:
I think it's ok with hard coded `FNV32A`. Thanks for updating!